web/src/settings_users.js

```diff
@@ -685,23 +702,31 @@ export function show_edit_user_info_modal(user_id, from_user_info_popover) {
             role: JSON.stringify(role),
             profile_data: JSON.stringify(profile_data),
         };
-        const opts = {
-            error_continuation() {
+        const $submit_btn = $("#edit-user-form .dialog_submit_button");
+        const $cancel_btn = $("#edit-user-form .dialog_exit_button");
+        show_button_spinner($submit_btn);
+        $cancel_btn.prop("disabled", true);
+
+        channel.patch({
+            url,
+            data,
+            success() {
+                user_profile.hide_user_profile();
```

Comment:
I think hiding the profile is probably the wrong feedback to give for a success -- it just doesn't feel nice and makes you reopen the widget to confirm your changes took effect. I think we might need to have a way to just show the successful save nicely, leaving the panel open.

Might want to be combined with having the "Save changes" button be disabled when nothing is different yet.